### PR TITLE
Doc: Add output to example code in programming FAQ

### PR DESCRIPTION
**Pull request by zhongjiajie**
Add output hint to document, part faq/programming, section [How do I write a function with output parameters (call by reference)?](https://docs.python.org/3/faq/programming.html#how-do-i-write-a-function-with-output-parameters-call-by-reference).

This patch make the output hint just like prefix code block.

Automerge-Triggered-By: @merwok

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@zhongjiajie

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by zhongjiajie** [comment]
I create new account in b.p.o, and singed CLA just now

**Comment by tirkarthi** [comment]
I have added skip-issue and skip-news labels since minor changes don't need issue and news entry.

**Comment by zhongjiajie** [comment]
@tirkarthi Thanks. Does it mean this PR could be merged after CLA pass/effective?

**Comment by tirkarthi** [comment]
It would still require an approval from a core developer to get merged. I added the labels so that the PR is green with all checks passing.

File: Doc/faq/programming.rst
Comment (merwok) [inline Doc/faq/programming.rst]: Would it be better to convert the snippet to console style?  Meaning

```python
>>> func3(args['a'], args['b'])
>>> print(args)
['new-value', 100]
```

File: Doc/faq/programming.rst
Comment (merwok) [inline Doc/faq/programming.rst]: hm that’s an ugly name

File: Doc/faq/programming.rst
Comment (merwok) [inline Doc/faq/programming.rst]: @serhiy-storchaka you added the `/` marker in #13720, but I don’t see why it’s useful (`__init__` being called implicitly with positional `self` param).  Mind if it’s removed?

Comment (serhiy-storchaka) [reply Doc/faq/programming.rst]: It is needed because you can pass keyword arguments with arbitrary names, and "self" is not an exception.
```python
args = callByRef(self=1)
assert args.self == 1
```
Without `/` the above example fails.

File: Doc/faq/programming.rst
Comment (zhongjiajie) [reply Doc/faq/programming.rst]: Should I change to console sytle?

File: Doc/faq/programming.rst
Comment (zhongjiajie) [reply Doc/faq/programming.rst]: @serhiy-storchaka Well but I test in my env, and get `SyntaxError`, did I miss something?
Screen Shot 2020-07-07 at 00 11 49

Comment (merwok) [reply Doc/faq/programming.rst]: You may be using an older Python version that doesn’t recognize it.
It is important that you use the right python to build the docs or check code — see https://devguide.python.org/documenting/#building-doc

File: Doc/faq/programming.rst
Comment (merwok) [reply Doc/faq/programming.rst]: That’s the question I asked.  What do you and @tirkarthi think?

Comment (zhongjiajie) [reply Doc/faq/programming.rst]: Well I think we should change to console style

Comment (tirkarthi) [reply Doc/faq/programming.rst]: I don't have a strong opinion on this. Other example codes have the output as comment like https://github.com/python/cpython/pull/21346/files#diff-986275b975a33c44c0aba973362516faL528 . So I am fine with either of the approach.

Comment (merwok) [reply Doc/faq/programming.rst]: Personally I don’t find output as comment very readable.
Let’s change it!

File: Doc/faq/programming.rst
Comment (zhongjiajie) [reply Doc/faq/programming.rst]: I change class name to `argSetter`, WDYT

File: Doc/faq/programming.rst
Comment (merwok) [inline Doc/faq/programming.rst]: This is simpler and shows the types more clearly:
```suggestion
      >>> func2(x, y)
      ('new-value', 100)
```

File: Doc/faq/programming.rst
Comment (merwok) [inline Doc/faq/programming.rst]:
```suggestion
      >>> func1(args)
      >>> args
      ['new-value', 100]
```

File: Doc/faq/programming.rst
Comment (merwok) [inline Doc/faq/programming.rst]:
```suggestion
      >>> args
      {'a': 'new-value', 'b': 100}
```

File: Doc/faq/programming.rst
Comment (merwok) [inline Doc/faq/programming.rst]: I think this should work:
```suggestion
      >>> vars(args)
      {'a': 'new-value', 'b': 100}
```

File: Doc/faq/programming.rst
Comment (merwok) [inline Doc/faq/programming.rst]: What do you think about this name: `Namespace`?

File: Doc/faq/programming.rst
Comment (merwok) [inline Doc/faq/programming.rst]: Funny that we have func2 before func1.  Switch them?

**Review by merwok — CHANGES_REQUESTED** [review]
Thanks for working on this improvement!

File: Doc/faq/programming.rst
Comment (zhongjiajie) [reply Doc/faq/programming.rst]: 👍

File: Doc/faq/programming.rst
Comment (zhongjiajie) [reply Doc/faq/programming.rst]: Yeah, it work 👍

File: Doc/faq/programming.rst
Comment (zhongjiajie) [reply Doc/faq/programming.rst]: I change it to `Namespace `

File: Doc/faq/programming.rst
Comment (zhongjiajie) [reply Doc/faq/programming.rst]: Fix

**Comment by zhongjiajie** [comment]
It's odd, this is a new commit in my branch https://github.com/zhongjiajie/cpython/commit/f71e07cc7456590116a5c7624a9139a81a94b254 but not including in this PR, I  don't know what going on.

**Review by merwok — APPROVED** [review]
Looks good!  I’d just like a second review.

**Comment by zhongjiajie** [comment]
@tirkarthi Could you please take a look at this patch? we need one more approval here

**Comment by zhongjiajie** [comment]
@tirkarthi Could you please take a look at this patch? we need one more approval here

**Review by tirkarthi — APPROVED** [review]
LGTM. Thanks.

**Comment by zhongjiajie** [comment]
So could we merge it to master?

**Comment by merwok** [comment]
I’m just not clear about backport rules given that 3.9 is in beta

**Comment by serhiy-storchaka** [comment]
Documentation changes can be backported to any maintained version. Feature freeze does not affect this.

**Comment by merwok** [comment]
Thanks for the PR!

**Pull request merged**
